dGame/EntityManager.cpp

```diff
@@ -161,111 +161,82 @@ void EntityManager::DestroyEntity(Entity* entity) {
 }
 
 void EntityManager::UpdateEntities(const float deltaTime) {
-	for (const auto& e : m_Entities) {
-	    e.second->Update(deltaTime);
+	for (auto entry = m_Entities.begin(); entry != m_Entities.end(); entry++) {
+		entry->second->Update(deltaTime);
 	}
 
-	for (const auto entityId : m_EntitiesToSerialize)
-	{
-		auto* entity = GetEntity(entityId);
+	for (auto entry = m_EntitiesToSerialize.begin(); entry != m_EntitiesToSerialize.end(); entry++) {
```

Comment:
`const auto& entry`. Given that you are de-referencing the entry below, I can't see this being more efficient than the short-hand form.

Reply:
having to de-reference it below due to entry being an iterator, which this is just how iterators work, as far as I know

Reply:
https://m.cplusplus.com/reference/vector/vector/push_back/

It's not about efficiency.  It's about adding to vectors when iterating through them as range based for loops.  When this is done, the end iterator is invalidated and since range based for loops are short hand for:
for(auto b = begin(), auto e = end(); b != e; ++b)
{
    auto value = *b;
}
There are cases where all iterators can be invalidated so it may be worth making this just a simple for loop in any case.